src/wallet/crypter.h

```diff
@@ -155,10 +158,19 @@ class CCryptoKeyStore : public CBasicKeyStore
             LOCK(cs_KeyStore);
             result = vMasterKey.empty();
         }
+        // fForMixing   fOnlyMixingAllowed  return
+        // ---------------------------------------
+        // true         true                result
+        // true         false               result
+        // false        true                true
+        // false        false               result
+
+        if(!fForMixing && fOnlyMixingAllowed) return true;
```

Comment:
The logic here seems very confusing and I don't think it's correct.

In particular if result is not empty I think this function should always return false, which your truth table doesn't guarantee.